tests/kani/AsyncAwait/manual_spawn.rs

```diff
@@ -2,22 +2,45 @@
 // SPDX-License-Identifier: Apache-2.0 OR MIT
 //
 // compile-flags: --edition 2018
-// kani-flags: --enable-unstable --mir-linker
 
 //! This file tests a hand-written spawn infrastructure and executor.
 //! This should be replaced with code from the Kani library as soon as the executor can get merged.
 //! Tracking issue: https://github.com/model-checking/kani/issues/1685
 
+use std::sync::{
+    atomic::{AtomicI64, Ordering},
+    Arc,
+};
+
 use std::{
     future::Future,
     pin::Pin,
-    sync::{
-        atomic::{AtomicI64, Ordering},
-        Arc,
-    },
     task::{Context, RawWaker, RawWakerVTable, Waker},
 };
 
+/// A very simple executor: it polls the future in a busy loop until completion
+///
+/// This is intended as a drop-in replacement for `futures::block_on`, which Kani cannot handle.
+/// Whereas a clever executor like `block_on` in `futures` or `tokio` would interact with the OS scheduler
+/// to be woken up when a resource becomes available, this is not supported by Kani.
+/// As a consequence, this function completely ignores the waker infrastructure and just polls the given future in a busy loop.
+///
+/// Note that spawn is not supported with this function. Use [`spawnable_block_on`] if you need it.
+// TODO: Give an error if spawn is used in the future passed to this function.
+pub fn block_on<T>(mut fut: impl Future<Output = T>) -> T {
```

Comment:
why not `kani::block_on`?

Reply:
see https://github.com/model-checking/kani/pull/1659#discussion_r1104463102

Reply:
removed as discussed above